common/utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any
+from dataclasses import dataclass
+from pathlib import Path
+
+from common.consts import MINIO_MAIN_PATH, PROJECT_PATH
+from metrics.consts import MetricCollections
+
+
+def singleton(cls) -> Any:
+    """Singleton implementation in form of decorator"""
+    instances = {}
+
+    def getinstance() -> Any:
+        if cls not in instances:
+            instances[cls] = cls()
+        return instances[cls]
+
+    return getinstance
+
+
+def create_weights_path(prefix: Path, collection_name: MetricCollections, suffix: str) -> Path:
+    return prefix / "data" / "models" / collection_name.value / suffix
+
+
+@dataclass
+class WeightsData:
+    """Dto containing all dataset information"""
+    collection_name: MetricCollections
```

Comment:
It this is the only parameter fullfiled, then you can remove all other members from dataclass and provide them as @property, this approach suggest that it has more than 1 field that can be changed

Reply:
Done